resources/source-types/sap_netweaver.yaml

```
@@ -0,0 +1,262 @@
+              - name: sapnetweaver.locks.enqueue.current.count
+              - name: sapnetweaver.locks.enqueue.high.count
+              - name: sapnetweaver.locks.enqueue.max.count
+              - name: sapnetweaver.locks.enqueue.lock_time
+              - name: sapnetweaver.locks.enqueue.lock_wait_time
+              - name: sapnetweaver.locks.enqueue.errors.count
+              - name: sapnetweaver.locks.dequeue.errors.count
+          - label: Connection
+            column: 1
+            metrics:
+              - name: sapnetweaver.connection.error.count
+          - label: Cache
+            column: 1
+            metrics:
+              - name: sapnetweaver.cache.hits
+              - name: sapnetweaver.cache.evictions
+          - label: Short Dumps
+            column: 1
+            metrics:
+              - name: sapnetweaver.short_dumps.rate
+      relevantIf:
+        - name: enable_metrics
+          operator: equals
+          value: true
+      advancedConfig: true
+
+    # Metrics TLS Settings #
+    - name: enable_tls
+      label: Enable TLS
+      description: Whether or not to use TLS.
+      type: bool
+      default: false
+      options:
+        sectionHeader: true
+      relevantIf:
+        - name: enable_metrics
+          operator: equals
+          value: true
+      advancedConfig: true
+
+    - name: insecure_skip_verify
+      label: Skip TLS Certificate Verification
+      description: Enable to skip TLS certificate verification.
+      type: bool
+      default: false
+      relevantIf:
+        - name: enable_tls
+          operator: equals
+          value: true
+      advancedConfig: true
+
+    - name: ca_file
+      label: TLS Certificate Authority File
+      description: Certificate authority used to validate TLS certificates.
+      type: string
+      required: false
+      default: ""
+      relevantIf:
+        - name: enable_tls
+          operator: equals
+          value: true
+      advancedConfig: true
+
+    - name: cert_file
+      label: Mutual TLS Client Certificate File
+      description: A TLS certificate used for client authentication, if mutual TLS is enabled.
+      type: string
+      required: false
+      default: ""
+      relevantIf:
+        - name: enable_tls
+          operator: equals
+          value: true
+      advancedConfig: true
+
+    - name: key_file
+      label: Mutual TLS Client Private Key File
+      description: A TLS private key used for client authentication, if mutual TLS is enabled.
+      type: string
+      required: false
+      default: ""
+      relevantIf:
+        - name: enable_tls
+          operator: equals
+          value: true
+      advancedConfig: true
+
+
+  metrics:
+    receivers: |
+      {{ if .enable_metrics }}
+      - sapnetweaver:
+          collection_interval: {{ .collection_interval }}s
+          endpoint: {{ .hostname }}:{{ .port }}
+          username: {{ .username }}
+          password: {{ .password }}
```

Sometimes fields like this can behave funny if not wrapped in single quotes.

```suggestion
          username: '{{ .username }}'
          password: '{{ .password }}'
```

---

updated